File: UnitTest.cpp
#include "Solution.hpp"
#include <iostream>

struct testcase {
    vector<int> arr;
    int output;
};

class UnitTest {
private:
    vector<testcase> testcases;
    Solution obj;
    
public:
    UnitTest() {
        testcases.push_back({{4,3,2,1,0}, 1});
        testcases.push_back({{1,0,2,3,4}, 4});
    }

    void test() {
        for(int i = 0; i < testcases.size(); ++i) {
            vector<int> arr = testcases[i].arr;
            int output = testcases[i].output;

            int result = obj.maxChunksToSorted(arr);
            cout << "Testcase " << i+1 << ": " << ((result == output) ? "passed":"failed") << endl;
        }
    }
};

int main() {
    UnitTest test;
    test.test();
}

File: UnitTest.py
from Solution import Solution
import unittest
from timeout_decorator import timeout

class UnitTest(unittest.TestCase):
    def setUp(self):
        self.__testcases = {1: ([4,3,2,1,0], 1), 2: ([1,0,2,3,4], 4)}
        self.__obj = Solution()
        return super().setUp()
    
    @timeout(0.5)
    def test_case_1(self):
        arr, output = self.__testcases[1]
        result = self.__obj.maxChunksToSorted(arr = arr)
        self.assertIsInstance(result, int)
        self.assertEqual(result, output)

    @timeout(0.5)
    def test_case_2(self):
        arr, output = self.__testcases[2]
        result = self.__obj.maxChunksToSorted(arr = arr)
        self.assertIsInstance(result, int)
        self.assertEqual(result, output)

if __name__ == '__main__': unittest.main()